Sources/MongoSwift/BSON/BSON.swift

```diff
@@ -307,20 +307,20 @@ extension BSON {
         BSONUndefined.self,
         MinKey.self,
         MaxKey.self,
-        Symbol.self,
+        BSONSymbol.self,
         Double.self,
         String.self,
         Document.self,
-        Binary.self,
-        ObjectID.self,
+        BSONBinary.self,
+        BSONObjectID.self,
```

Comment:
I feel like this is the most bikeshedy of bikeshedy territories, but a few things occur to me:
- `ObjectID` is unlikely to clash
- You've omitted `MinKey` and `MaxKey` which seem like they have equal "clash quotient"
- `Document` actually has quite a high "clash quotient" IMO
- I really dislike all the `BSON` prefixes on paper, but I think it may be argued similar to the JavaScript story that you're unlikely to really ever see these types in your daily programming. If we're prefixing some of them, should we just prefix _all_ of them? 

Does clashing even matter? Something like `Index` => `IndexHint`, or `Server` => `ServerAddress` are acceptable changes because they provide more contextual information while reducing the potential for collisions. Does the same hold true for `Timestamp`? Maybe.

Reply:
>If we're prefixing some of them, should we just prefix all of them?

I also dislike the prefixing on paper, but I tend to agree with this. I think it could be surprising for users that some types aren't prefixed, even if those types are very unlikely to clash.

>Does clashing even matter?

Unfortunately the clashing matters a lot. If a user happens to add a new dependency that clashes, all their old code breaks and will need to be updated to be namespace prefixed. Worse yet, if a clashing type is adding to an _existing_ dependency in a minor version, it could break existing code.

e.g. The following would break if B introduces `Dog` in 1.1.
```swift
import A // 1.0
import B // 1.0

func main() {
    let x = Dog() // From A
}
```

Reply:
`MinKey` and `MaxKey` are `internal` types - they're just a place for the decoding/encoding logic for those to live, but users only interact with them via the `BSON` enum cases `.minKey` and `.maxKey`. So I didn't bother with prefixing those as the clashing issue doesn't apply, but we could prefix them for consistency internally anyway.

agree `Document` is one that is likely to clash - as I noted in the description I just left that off of this PR as it's a huge diff and I figured worth putting into a follow-up PR.

I'm fine with the all or nothing approach to prefixing on public types anyway.

yeah, the clashing issue is pretty frustrating, and makes semver feel a lot less useful. it's impossible to solve completely as someone *could* theoretically reuse any type name you come up with, no matter how unique, but I think we have to at least do our best to prevent issues with it.

Reply:
OK, prefix all the things it is I guess :man_facepalming:

Reply:
kk, I'll do the remaining types in a second PR